Sources/ArgumentParser/Parsable Types/ParsableArgumentsValidation.swift

```diff
@@ -185,3 +186,63 @@ struct ParsableArgumentsCodingKeyValidator: ParsableArgumentsValidator {
     }
   }
 }
+
+/// Ensure argument names are unique within a `ParsableArguments` or `ParsableCommand`.
+struct ParsableArgumentsUniqueNamesValidator: ParsableArgumentsValidator {
+  struct Error: Swift.Error, CustomStringConvertible {
+    var duplicateNames: [String: Int] = [:]
+
+    var occurred: Bool {
+      !duplicateNames.isEmpty
+    }
+
+    var description: String {
+      var description = duplicateNames.reduce(into: "") { description, entry in
+        description += "Multiple (\(entry.value)) `Option` or `Flag` arguments are named \"\(entry.key)\".\n"
+      }
+      if description.last == "\n" {
+        description.removeLast()
+      }
+      return description
+    }
+  }
+
+  static func validate(_ type: ParsableArguments.Type) throws {
+    let argSets: [ArgumentSet] = Mirror(reflecting: type.init())
+      .children
+      .compactMap { child in
+        guard
+          var codingKey = child.label,
+          let parsed = child.value as? ArgumentSetProvider
+          else { return nil }
+
+        // Property wrappers have underscore-prefixed names
+        codingKey = String(codingKey.first == "_" ? codingKey.dropFirst(1) : codingKey.dropFirst(0))
+
+        let key = InputKey(rawValue: codingKey)
+        return parsed.argumentSet(for: key)
+    }
```

Comment:
The validation is kicked off in that particular `ArgumentSet` initializer, so using it in a validator causes infinite recursion. I think that's why the reflection code is copied from the initializer into each of the validators. That's the duplication I mentioned I'd like to factor out.

I have a rough idea that could be done by converting the validators to functors with signatures like `(ArgumentSet) throws -> Void` (except that `ParsableArgumentsCodingKeyValidator` needs to be something like `([String]) throws -> Void` because it iterates over the coding keys). Then those functors could be called from the points in the reflection code where the coding keys or argument sets they need are available (for debug only, as currently). But I need to think about that more, because I'm not sure whether there are other validators we'd want that would have different signatures. I wouldn't want to have an explosion of validator types and places they're being called from.

Reply:
Other possible approaches would be:
- Moving the `_validate()` method to another location. 
- Moving the duplicated code to a helper function.